File: java/resistor-color/src/main/java/ResistorColor.java
import java.util.Arrays;

class ResistorColor {

    private enum Colors {
        BLACK,
        BROWN,
        RED,
        ORANGE,
        YELLOW,
        GREEN,
        BLUE,
        VIOLET,
        GREY,
        WHITE;

        @Override
        public String toString() {
            return this.name().toLowerCase();
        }
    }

    int colorCode(String color) {
        return Colors.valueOf(color.toUpperCase()).ordinal();
    }

    String[] colors() {
        return Arrays.stream(Colors.values())
                .map(Colors::toString)
                .toArray(String[]::new);
    }
}

File: dart/resistor-color/lib/resistor_color.dart
class ResistorColor {
  static const resistorCodes = {
    'black': 0,
    'brown': 1,
    'red': 2,
    'orange': 3,
    'yellow': 4,
    'green': 5,
    'blue': 6,
    'violet': 7,
    'grey': 8,
    'white': 9,
  };

  int colorCode([String color = 'black']) =>
      resistorCodes[color] ?? -1;

  List<String> get colors => resistorCodes.keys.toList();
}
